sopel/modules/currency.py

```diff
@@ -79,10 +79,17 @@ def build_reply(amount, base, target, out_string):
     rate = float(rate_raw)
     result = float(rate * amount)
 
-    if target == 'BTC':
-        return out_string + ' {:.5f} {},'.format(result, target)
+    digits = 0
+    # up to 10 (8+2) digits precision when result is less than 1
+    # as smaller results need more precision
+    while digits < 8 and 1 / 10**digits > result:
+        digits += 1
```

Comment:
What if instead of `while`, something like:

```suggestion
    # smaller results (usually cryptocurrencies) need more precision
    whole, decimal = repr(result).split('.')
    digits = min(1, len(whole))
    digits += len(decimal) - len(decimal.lstrip('0'))
```

Not sure how it would handle roundoff errors, though. Some (e.g. the classic 0.30000000000000004) would be fine, but surely it would give weird results if Python is capable of e.g. 0.1 turning into 0.0999.

Just a thought. @Exirel mentioned the `while` loop isn't his favorite, so I did my best to come up with _an idea_ even if we end up not using it.

Reply:
While I agree the un-commented while loop can be confusing, I think with a bit of an explanation of the intuitive way it works could help that. Not sure if I like that there's more comments than code tho :P 
```suggestion
    
    # up to 10 (8+2) digits precision when result is less than 1
    # as smaller results (usually cryptocurrencies) need more precision
    # Move the decimal point to the right by dividing by 10
    # Stop when we have enough digits to show something in result, or we've hit our limit
    digits = 0
    while digits < 8 and 1 / 10**digits > result:
        digits += 1
```
Also, it seems repr returns engineering notation on very small numbers
```
>>> repr(0.000001)
'1e-06'
```

Reply:
Sure would be nice to find a constant-time algorithm, though, even if the O(n) one is limited to 8 iterations. In this case it's the principle, not the actual performance impact, that has _me_ interested.

> Also, it seems repr returns engineering notation on very small numbers

Annoying. I guess [one](https://stackoverflow.com/a/38847691/5991) of [these](https://stackoverflow.com/a/38983595/5991) would be needed to stringify the number as expected, but at that point it's probably more efficient to do it another way.